loaders/src/glTF/2.0/Extensions/KHR_mesh_instancing.ts

```diff
@@ -0,0 +1,111 @@
+                    let attributeKey = attributeKeys[i];
+                    let accessor = ArrayItem.Get(`${extensionContext}/attributes/${attributeKey}`, this._loader.gltf.accessors, attributes[attributeKey]);
+                    attributeBufferPromises.push(this._loader._loadFloatAccessorAsync(`/accessors/${accessor.bufferView}`, accessor).then((data) => {
+                        attributeBuffers[attributeKey] = { buffer : data, accessor : accessor };
+                    }));
+                }
+
+                const mesh = ArrayItem.Get(`${context}/mesh`, this._loader.gltf.meshes, extension.mesh || nodeMesh);
+                this._loader._loadMeshAsync(`/meshes/${mesh.index}`, node, mesh, (babylonMeshInstanceNode) => {
+                    Promise.all(attributeBufferPromises).then(() => {
+                        attributeKeys = Object.keys(attributeBuffers);
+
+                        if (attributeKeys.length > 0) {
+                            let instanceCount = attributeBuffers[attributeKeys[0]].accessor.count;
+                            let digitLength = instanceCount.toString().length;
+                            for (let i = 0; i < instanceCount; ++i) {
+                                let instance;
+                                if (i == 0) {
+                                    instance = babylonMeshInstanceNode;
+                                    instance.setParent(babylonTransformNode);
+                                } else {
+                                    instance = babylonMeshInstanceNode.clone((babylonMeshInstanceNode.name || babylonMeshInstanceNode.id) + "_" + String(i).padStart(digitLength, '0'), babylonTransformNode, true);
+                                }
+
+                                if (instance) {
+                                    instance.position = attributeBuffers["TRANSLATION"] ? Vector3.FromArray(attributeBuffers["TRANSLATION"].buffer, i * 3)
+                                                                                        : Vector3.Zero();
+                                    instance.rotationQuaternion = attributeBuffers["ROTATION"] ? Quaternion.FromArray(attributeBuffers["ROTATION"].buffer, i * 4)
+                                                                                            : Quaternion.Identity();
+                                    instance.scaling = attributeBuffers["SCALE"] ? Vector3.FromArray(attributeBuffers["SCALE"].buffer, i * 3)
+                                                                                    : Vector3.One();
+                                    instance.computeWorldMatrix(true);
+                                }
+                            }
+                        }
+                        let babylonMesh = (babylonTransformNode as Mesh);
+                        if (babylonMesh) {
+                            babylonMesh.isVisible = false;
+                        }
```

Comment:
I'm not sure about this. I think perhaps the first one should be a regular mesh and the other ones will be instances of the first one to avoid adding an invisible source mesh? There seems to already be a check for `i == 0`, which I'm also not a big fan of.

Reply:
I agree, this has been modified to instead iterate through all the mesh primitives, and use the transformNodes returned via `loadNodeAsync` as our source meshes to clone for each primitive instance.